functions.sh

```diff
@@ -114,15 +117,31 @@ tpm2_extend() {
 tpm2_flushall() {
 	tpm2 flushcontext \
 		--transient-object \
-	|| die "tpm2_flushcontext: unable to flush transient handles"
+	|| die "tpm2_flushcontext: unable to flush transient handles" \
```

Comment:
Seeing any code after `die` confuses my Perl brain, so I think we should have `die` *always* exit, and for these helper functions we should use a separate `error` function (with the `|| error foo || return 1` template).  That way it is clear that `die` will really kill things (and the callers should be checking in case they have any clean up anyway)

Reply:
Fixed.